src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -99,7 +99,8 @@ protected override ImmutableArray<LocalSymbol> BuildLocals()
                 case SyntaxKind.IdentifierName:
                     break;
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(declaration.Kind());
+                    ExpressionVariableFinder.FindExpressionVariables(this, locals, declaration);
```

Comment:
>ExpressionVariableFinder.FindExpressionVariables(this, locals, declaration); [](start = 20, length = 76)

Could you please add a comment about the scenario when we get here and point me to the test(s) that covers this line of code?